storage/remote/queue_manager.go

```diff
@@ -904,10 +1017,15 @@ func (s *shards) runShard(ctx context.Context, shardID int, queue chan sample) {
 	// If we have fewer samples than that, flush them out after a deadline
 	// anyways.
 	var (
-		max            = s.qm.cfg.MaxSamplesPerSend
-		nPending       = 0
-		pendingSamples = allocateTimeSeries(max)
-		buf            []byte
+		max = s.qm.cfg.MaxSamplesPerSend
+		// Rough estimate, 1% of active series will contain an exemplar on each scrape.
```

Comment:
This is to expand the amount of data that can be buffered by a small amount to try and account for exemplars being present but still achieve the same throughput for samples data, rather than buffering samples and exemplars together and having exemplars take slots away from samples. We're not entirely there with the current implementation, so I'll file an issue to improve the buffering logic and also add a note to the configuration documentation.

Reply:
👍🏽 thanks!